### DISCUSSION TITLE
Does forklift support K8s 1.22

### DISCUSSION THREAD

**Discussion by lxs137** [kubev2v/forklift#291] [Q&A] [answered]
Does forklift v2.4.0 support K8s 1.22, I am confused with docs

1. in docs, we say "Kubernetes cluster or Minikube v1.22+": https://github.com/kubev2v/forklift/blob/main/operator/docs/k8s.md#pre-requisites

2. but in ClusterServiceVersion/forklift-operator.v2.4.0, we say "minKubeVersion: 1.24.0": [https://github.com/kubev2v/forklift/blob/df08a623dcbc5a23d1337f6001332ca37883e027/[…]ig/manifests/bases/forklift-operator.clusterserviceversion.yaml](https://github.com/kubev2v/forklift/blob/df08a623dcbc5a23d1337f6001332ca37883e027/operator/config/manifests/bases/forklift-operator.clusterserviceversion.yaml#L156)

**Comment by ahadas** [comment, marked as answer]
@lxs137 yeah, the prerequisites should have been updated with 1.24+.
Besides the fact that forklift v2.4.0 wasn't tested with Kubernetes < 1.24, we've also started to use the 'Volume Populators' feature in forklift v2.4.0 which exists in Kubernetes 1.22 but requires to enable the feature gate `AnyVolumeDataSource` that is disabled by default. That said, it could well be that forklift v2.4.0 would work on Kubernetes 1.22 if you don't initiate migrations from OpenStack or cold-migrations from RHV, or enable the `AnyVolumeDataSource` feature gate - but it's recommended to use it with Kubernetes 1.24+.
